src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -62,19 +77,396 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
     }
   }
 
-  private Future<Void> savePermList(List<Perm> permList, Context vertxContext, String tenantId) {
+  private Future<List<Permission>> getPermsForModule(ModuleId moduleId, List<OkapiPermission> permSet,
+      Context vertxContext, String tenantId) {
+    Promise<List<Permission>> promise = Promise.promise();
+    
+    try {
+      getPermsByModule(moduleId, vertxContext, tenantId)
+        .onComplete(getResult -> {
+          if (getResult.failed()) {
+            Throwable cause = getResult.cause();
+            logger.error(cause.getMessage(), cause);
+            promise.fail(cause);
+            return;
+          }
+
+          List<Permission> existing = getResult.result();
+          if (existing.isEmpty() && permSet != null) {
+            // this means either:
+            // A) the first time enabling this module, or
+            // B) the permissions exist but don't yet have the definedBy field
+            List<OkapiPermission> perms = permSet.stream()
+                .filter(p -> p.getPermissionName() != null)
```

Comment:
I see this filter is used a lot. Could there be a permission without name?

Reply:
Good catch.
Its description says: "The unique permission name".
It should be a required non-null property.

Reply:
Just doing it once in `handlePermLists(...)` now.  See https://issues.folio.org/browse/OKAPI-974